src/uint/div_limb.rs

```diff
@@ -145,43 +145,49 @@ pub(crate) const fn div2by1(u1: Word, u0: Word, reciprocal: &Reciprocal) -> (Wor
     (q1, r)
 }
 
-/// Calculate the quotient and the remainder of the division of a 3-word
-/// dividend `u`, with a precalculated leading-word reciprocal `v` and a second
-/// divisor word `v0`. The dividend and the lower divisor word must be left-shifted
-/// according to the `shift` value  of the reciprocal.
+/// Given two long integers `u = (..., u0, u1, u2)` and `v = (..., v0, v1)`
+/// (where `u2` and `v1` are the most significant limbs), where `floor(u / v) <= Limb::MAX`,
+/// calculates `q` such that `q - 1 <= floor(u / v) <= q`.
+/// In place of `v1` takes its reciprocal, and assumes that `v` was already pre-shifted
+/// so that v1 has its most significant bit set (that is, the reciprocal's `shift` is 0).
 #[inline(always)]
-pub(crate) const fn div3by2(
+pub(crate) const fn div3by2_vartime(
```

Comment:
I'm also worried about the runtime of this method depending on the dividend value. Replacing it completely would also mess with my PR #643, do you want to do a quick review of that one?

Reply:
The original `div3by2()` can be preserved, of course, but I am worried about the connection of its contents and the original algorithm from the article. Could you add some explanations about why it actually does the same thing?

Reply:
I added a PR to better align that method with the definition, as it doesn't look like overflow was being handled properly in the remainder.

Reply:
@andrewwhitehead can you take a look again now that #643 is merged?